tests/simpleKVBC/cmf/skvbc_messages.cmf

```diff
@@ -0,0 +1,86 @@
+# Definition of client-server interface for the SimpleKVBC test application.
+#
+# Note SimpleKVBC effectively implements a versioned key-value store supporting
+# only keys and values that are byte strings of exactly 21 bytes.
+#
+# Note that, when key-value pairs (of 21-byte byte strings) need to be handled
+# in this CMF interface, we use length-2 fixed lists of length-21 fixed lists of
+# 8-bit signed integers. While CMF natively supports key-value pairs with its
+# kvpair type, we choose not to use it for key-value pairs in the SimpleKVBC CMF
+# interface because CMF's native kvpair type only supports CMF's primitive types
+# as keys. Though CMF's string and/or bytes primitive types could be used to
+# encode our length-21 byte strings for SimpleKVBC, we would prefer these byte
+# strings always be encoded with CMF's fixedlist type so that CMF can enforce
+# SimpleKVBC's exact length requirements for its keys and values.
+
+Msg SKVBCReadRequest 3 {
+
+    # Note an SKVBCReadRequest with read version 0 indicates that the latest
+    # version should be read.
+    uint64 read_version
+
+    list fixedlist uint8 21 keys
+}
+
+Msg SKVBCWriteRequest 4 {
+    uint64 read_version
+
+    # if set to true, the SimpleKVBC replica will waste some time when it works
+    # on pre-processing this request, which may be useful for simulating certain
+    # workloads in testing.
+    bool long_exec
+
+    list fixedlist uint8 21 readset
+    list fixedlist fixedlist uint8 21 2 writeset
+}
+
+Msg SKVBCGetLastBlockRequest 5 {
+
+    # Though this message does not need to convey any information beyond that
```

Comment:
This is definitely a limitation of CMF right now, although it's likely easily solved. I believe you may only need to change a `+` to a `*` in the [grammar](https://github.com/vmware/concord-bft/blob/570bd6fad4e54f89ce4910107b3e689be4adf869/messages/compiler/grammar.ebnf#L8). It may also need some small fixes for code generation, and at a minimum tests for both C++ and Python.

Reply:
I have attempted to remove this limitation as you suggested in my most recent revisions to this PR; I have changed the `+` to an `*`, removed this unused field from the `SKVBCGetLastBlockRequest`. I have also attempted to revise the CMF code generation to fix it so that it runs, even with empty messages, so that the generated 
C++ compiles, and so that a generated Python file can be run without being rejected for syntax errors or other such malformations. I have also validated that `test_cppgen.py` still appears to work. Are there any other revisions I ought to add to CMF to accommodate and test this support for empty messages?

Reply:
That should be good enough! Thanks Alex.